backend/cmd/headlamp.go

```diff
@@ -960,7 +992,92 @@ func (c *HeadlampConfig) startPortForward(p PortForwardPayload, token string) er
 	return nil
 }
 
-func (c *HeadlampConfig) handleClusterRequests(router *mux.Router) {
+// Returns the helm.Handler given the config and request. Writes http.NotFound if clusterName is not there.
+func getHelmHandler(c *HeadlampConfig, w http.ResponseWriter, r *http.Request) (*helm.Handler, error) {
+	clusterName := mux.Vars(r)["clusterName"]
+	context, ok := c.contextProxies[clusterName]
+
+	if !ok {
+		http.NotFound(w, r)
+		return nil, errors.New("not found")
+	}
+
+	namespace := r.URL.Query().Get("namespace")
+
+	helmAction, err := helm.NewActionConfig(context.context.clientConfig(), namespace)
+	if err != nil {
+		log.Printf("Error: failed to create helm action config: %s", err)
+		http.Error(w, "failed to create helm action config", http.StatusInternalServerError)
+	}
+
+	helmHandler := helm.Handler{Configuration: helmAction}
+
+	return &helmHandler, nil
+}
+
+//nolint:gocognit
+func handleClusterHelm(c *HeadlampConfig, router *mux.Router) {
+	router.PathPrefix("/clusters/{clusterName}/helm/{.*}").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		helmHandler, err := getHelmHandler(c, w, r)
+		if err != nil {
+			return
+		}
+
+		// we used nolint:gocognit in this function because...
+		//  Perhaps there's a better way to dispatch these?
```

Comment:
Is there a better way to dispatch these?

Reply:
This is fine to improve later IMO.

Reply:
Due to the current structure of backend I couldn't find better ways to call the helm handler functions. Looking into ways to improve this and the URL structure for helm endpoints.